File: finger_exercises/biggest_uneven.py
""" Write a program that examines three variables (x,y,z) and
prints the largest odd number among them. If none are odd, print the smallest value"""

data = input("Please enter three numbers, followed by a space: ")
nums = data.split(" ")
x, y, z = [int(x) for x in nums]

ans = min(x,y,z)
if x % 2 != 0:
    ans = x
if y % 2 != 0 and y > ans:
    ans = y
if z % 2 != 0 and z > ans:
    ans = z

print(ans)

File: finger_exercises/biggest_uneven.c
#include <stdio.h>

int main(void){
    int nums[3];
    int ans;
    for (int i = 0; i < 3; i++){
        printf("Enter an integer: ");
        scanf("%d", &nums[i]);
    }

    ans = nums[0];
    for (int i = 1; i < 2; i++)
    {
        if (nums[i] < ans)
            ans = nums[i];
    }
    

    if (nums[0] % 2 != 0)
        ans = nums[0];
    if ((nums[1] % 2 != 0) && (nums[1] > nums[0]))
        ans = nums[1];
    if ((nums[2] % 2 != 0) && (nums[2] > nums[1]))
        ans = nums[2];
    printf("%d\n", ans);

    return 0;
}